### PR TITLE
datagenerator is now up to date

### LINKED ISSUE #104: update the dataGen script
add new functionality to the dataGen

### PR DESCRIPTION
**Pull request by esteeroramy**
fixes #104 
This generator is not up to date with all the functionality that we have set in the website so far.

### CODE REVIEW

File: contents.js [hunk 1]
```diff
@@ -0,0 +1,504 @@
+exports.namesList = [
```
Comment (ahmedqarmout2) [inline contents.js]: change the name of the file please. something like namesDictionary.js

Comment (esteeroramy) [reply contents.js]: It does not only include names though

File: contents.js [hunk 2]
```diff
@@ -0,0 +1,504 @@
...
+'Calvin Blanchard',
+'Madalynn Flores',
+'Kaydence Horton',
+'Demarion Anderson',
+'Gilberto Moore',
+'Kyan Farmer',
+'Braylon Tyler'
+]
+
+exports.comment = 'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Duis iaculis auctor sollicitudin. Nam volutpat libero ut massa placerat efficitur. Pellentesque lobortis consectetur fermentum. Nullam pharetra ac elit nec accumsan. Praesent eu diam a arcu fermentum efficitur eu nec ipsum. In blandit arcu a ipsum consectetur posuere. Sed vitae elit tempor, dapibus diam et, eleifend tellus. Pellentesque habitant morbi tristique senectus et netus et malesuada fames ac turpis egestas. Aliquam erat volutpat. Sed id pulvinar dui. Pellentesque nec vehicula ex.'
```
Comment (ahmedqarmout2) [inline contents.js]: now that name doesnt make sense

Comment (ahmedqarmout2) [reply contents.js]: plz figure out a better name for the file :)

Comment (esteeroramy) [reply contents.js]: LOL

File: dataGenerator.js [hunk 3]
```diff
@@ -23,149 +23,931 @@ var db = require('./server/db.js');
 var users = require('./server/users.js');
 var questions = require('./server/questions.js');
 var common = require('./server/common.js');
+var contents = require('./contents.js');
 
 // variables to control the genereated data
+var numberOfEachQuestion = [2, 2, 2, 2, 2, 2];
```
Comment (ahmedqarmout2) [inline dataGenerator.js]: a dictionary would be nice here. really hard to tell which number is for which question

File: dataGenerator.js [hunk 4]
```diff
@@ -23,149 +23,931 @@ var db = require('./server/db.js');
...
+var commentActionsPerQuestion = 3;
+var commentRepliesVotesPerQuestion = 3;
+
+// Probabilities used in generating data
+var commentPercentage = 50;
+var commentActionPercentage = 50;
+var mentionsPercentage = 50;
 var questionsCorrectPercentage = 40;
 
+
```
Comment (ahmedqarmout2) [inline dataGenerator.js]: extra space.exe?

File: contents.js (hunk 1 above)
Comment (esteeroramy) [reply contents.js]: name has been changed

File: dataGenerator.js (hunk 3 above)
Comment (esteeroramy) [reply dataGenerator.js]: done

File: dataGenerator.js (hunk 4 above)
Comment (esteeroramy) [reply dataGenerator.js]: removed

File: dataGenerator.js [hunk 5]
```diff
@@ -174,42 +965,90 @@ var checkAnswer = function (questionId, userId, answer, callback) {
...
 }
 
-var createQuestions = function() {
-  	for (var id = 1; id <= questionsCount; id++) {
-      	addQuestion('Is math related to science? '+id, id);
-  	}
+/**
+ * This function resets the variables that are needed for each question to be pupulated
+ * 
+ * @param {integer} number
```
Comment (ahmedqarmout2) [inline dataGenerator.js]: is this still a number

Comment (esteeroramy) [reply dataGenerator.js]: nice catch

Comment (esteeroramy) [reply dataGenerator.js]: done

**Review by Mohammed-JS-Khan — APPROVED** [review]
Lorem ipsum dolor sit amet, consectetur adipiscing elit. Duis iaculis auctor sollicitudin. Nam volutpat libero ut massa placerat efficitur. Pellentesque lobortis consectetur fermentum. Nullam pharetra ac elit nec accumsan. Praesent eu diam a arcu fermentum efficitur eu nec ipsum. In blandit arcu a ipsum consectetur posuere. Sed vitae elit tempor, dapibus diam et, eleifend tellus. Pellentesque habitant morbi tristique senectus et netus et malesuada fames ac turpis egestas. Aliquam erat volutpat. Sed id pulvinar dui. Pellentesque nec vehicula ex.

File: dataGenerator.js [hunk 6]
```diff
@@ -23,149 +23,940 @@ var db = require('./server/db.js');
 var users = require('./server/users.js');
 var questions = require('./server/questions.js');
 var common = require('./server/common.js');
+var datagenInfo = require('./datagenInfo.js');
 
 // variables to control the genereated data
+
```
Comment (ahmedqarmout2) [inline dataGenerator.js:29]: extraSpace.html

Comment (esteeroramy) [reply dataGenerator.js:29]: dkm, fixed

Comment (ahmedqarmout2) [reply dataGenerator.js:29]: this is no laughing matter. This is serious

Comment (ahmedqarmout2) [reply dataGenerator.js:29]: also can we add animation

Comment (esteeroramy) [reply dataGenerator.js:29]: consider it done

Comment (ahmedqarmout2) [reply dataGenerator.js:29]: nice work fam. I knew I can always count on you

**Review by mehtadarshan1 — APPROVED** [review]
good job

File: dataGenerator.js [hunk 7]
```diff
@@ -23,149 +23,939 @@ var db = require('./server/db.js');
...
+            for (var i = 0; i < adminsCount; i++) {
+                rateQuestion(id, 'Admin' + i, Math.floor(Math.random() * 6));
+            }
+        }
+
+        if (questionsCreated === numberOfQuestionsExpected) {
+            answerQuestionsMultipleChoice();
+        }
+
+        questionsCreated++;
```
Comment (ahmedqarmout2) [inline dataGenerator.js:281]: since the counts start at 0, should we check the if statement after the counting?

Comment (esteeroramy) [reply dataGenerator.js:281]: that is how it used to be previously, and there is a delayed call that used to run after that if statement, this fixes that issue

File: dataGenerator.js [hunk 8]
```diff
@@ -23,149 +23,939 @@ var db = require('./server/db.js');
...
+        } else if (!question) {
+            return callback('Could not find the question', null);
+        } else {
+            var answeredList = [];
+
+            for (var i in question.correctAttempts) {
+                answeredList.push(question.correctAttempts[i].id);
+            }
+
+            for (var j = 0; j < question.comments.length; j++) {
```
Comment (ahmedqarmout2) [inline dataGenerator.js:891]: this is so efficient. Can we use some dynamic programming to optimise this?

Comment (esteeroramy) [reply dataGenerator.js:891]: nopenopenopenope

**Review by ahmedqarmout2 — APPROVED** [review]

**Pull request merged**